pkg/config/configuration.go

```diff
@@ -123,3 +138,26 @@ func LoadKubernetesConfiguration(config, namespace string, operatorClient operat
 	}
 	return &conf, nil
 }
+
+func IsSecretAllowed(storeName, key, defaultSecretAccess string, allowedSecrets, deniedSecrets map[string]map[string]struct{}) bool {
```

Comment:
If we can extract the triple `(defaultAccess, allowedSecrets, deniedSecret)` into a struct we can move this method to that struct. It seems in our function calls these three go together a lot so I think we will probably save some code and may be avoid cases where we may mistakenly swap `(allowed, denied)`.